File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
  let w = 0;
  let h = 0;
    sizes.map((x)=>x.sort((a,b)=>a-b)).map((x)=>{
        w = Math.max(w,x[0]);
        h = Math.max(h,x[1]);
    })
    return w*h
}

         

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int row = 0;
        int col = 0;
        
        for(int i = 0; i<sizes.length; i++){
            int w = Math.min(sizes[i][0],sizes[i][1]);
            int h = Math.max(sizes[i][0],sizes[i][1]);
            
            row = Math.max(row, w);
            col = Math.max(col, h);
        }
    
        
        answer = row*col;
        return answer;
    }
}